IdentitySdkCore/IdentitySdkCore/Classes/ReachFiveFidoClient.swift

```
@@ -0,0 +1,179 @@
+          
+           let requireResidentKey = true
+           var options = PublicKeyCredentialCreationOptions()
+           challenge = base64urlToHexString(base64url: challenge)
+          
+           options.challenge = Bytes.fromHex(challenge)
+           options.user.id = Bytes.fromString(userId)
+           options.user.name = userName
+           options.user.displayName = displayName
+           options.rp.id = rpId
+           options.rp.name = rpId
+          
+           options.attestation = .direct
+           options.addPubKeyCredParam(alg: .es256)
+           options.authenticatorSelection = AuthenticatorSelectionCriteria(
+               requireResidentKey: requireResidentKey,
+               userVerification: UserVerificationRequirement.required
+           )
+
+           firstly {
+               self.webAuthnClient.create(options)
+           }.done { credential in
+               
+               let rawId = credential.rawId.toHexString()
+               let credId = credential.id
+               let credType = credential.type.rawValue
+               let clientDataJSON = self.encodeClientDataJsonToBase64(clientDataJson: credential.response.clientDataJSON)
+               let attestationObject = Base64.encodeBase64URL(credential.response.attestationObject)
+
+               let r5AuthenticatorAttestationResponse = R5AuthenticatorAttestationResponse(attestationObject: attestationObject,clientDataJSON: clientDataJSON)
+               
+               let registrationPublicKeyCredential = RegistrationPublicKeyCredential(id: credId,rawId: rawId,type: credType,response: r5AuthenticatorAttestationResponse)
+               
+               let webauthnSignupCredential = WebauthnSignupCredential (webauthnId: userId,publicKeyCredential: registrationPublicKeyCredential)
+               _ = completion(webauthnSignupCredential)
+
+           }.catch { error in
+               print("error : ",error.localizedDescription)
```

Will this swallow the error that should instead be thrown for the integrator to catch or handle?

---

For synchronous calls like this method a `Result<T, E>` could be returned instead so the error handling should be done by the caller. For async calls `Future<T, E>` should be used.